pubsub/topics/publish_settings.go

```diff
@@ -19,34 +19,56 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"strconv"
+	"sync"
+	"sync/atomic"
 	"time"
 
 	"cloud.google.com/go/pubsub"
 )
 
-func publishWithSettings(w io.Writer, projectID, topicID, msg string) error {
+func publishWithSettings(w io.Writer, projectID, topicID string) error {
 	// projectID := "my-project-id"
 	// topicID := "my-topic"
-	// msg := "Hello World"
 	ctx := context.Background()
 	client, err := pubsub.NewClient(ctx, projectID)
 	if err != nil {
 		return fmt.Errorf("pubsub.NewClient: %v", err)
 	}
 
+	var wg sync.WaitGroup
+	var totalErrors uint64
 	t := client.Topic(topicID)
 	t.PublishSettings.ByteThreshold = 5000
 	t.PublishSettings.CountThreshold = 10
 	t.PublishSettings.DelayThreshold = 100 * time.Millisecond
 
-	result := t.Publish(ctx, &pubsub.Message{Data: []byte(msg)})
-	// Block until the result is returned and a server-generated
-	// ID is returned for the published message.
-	id, err := result.Get(ctx)
-	if err != nil {
-		return fmt.Errorf("Get: %v", err)
+	for i := 0; i < 10; i++ {
+		result := t.Publish(ctx, &pubsub.Message{
+			Data: []byte("Message " + strconv.Itoa(i)),
+		})
+
+		wg.Add(1)
```

Comment:
What if we appended the results to a slice, then iterate through that slice calling `res.Get`? If we go that route, I don't think we need any goroutines, waitgroup, or atomic.

We can also append errors to a slice or only keep the last non-nil error.